src/macros.rs

```diff
@@ -0,0 +1,278 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[macro_export]
+macro_rules! labels {
+    () => {
+        {
+            use std::collections::HashMap;
+
+            HashMap::new()
+        }
+    };
+
+    ( $ ( $ KEY : expr => $ VALUE : expr , ) + ) => {
+        {
+            use std::collections::HashMap;
+
+            let mut lbs = HashMap::new();
+            $(
+                lbs.insert($KEY, $VALUE);
+            )+
+
+            lbs
+        }
+    }
+}
```

Comment:
Consider `labels!{}`, then `mut` is unnecessary and clippy complains [`let_and_return`](https://github.com/Manishearth/rust-clippy/wiki#let_and_return)